src/components/Header.tsx

```diff
@@ -1,20 +1,22 @@
 declare let window: any;
 import React, { FunctionComponent, useEffect, useState } from "react";
-import styled from "@emotion/styled";
-import { OrangeButton } from "./common/Button";
-import logo from "../images/logo.svg";
+import { OrangeButton } from "./common";
 import { getWalletDetails as getWalletData } from "./util/wallet";
-import { getEtherscanAddress } from "./util/util";
+import { getEtherscanAddress } from "./util/common";
+import logo from "../images/logo.svg";
+import styled from "@emotion/styled";
 
 const BreakLine = styled.div`
   flex-basis: 100%;
 `;
 
 export const Header: FunctionComponent = () => {
   const [isConnected, setIsConnected] = useState(false);
-  const [walletAddress, setWalletAddress] = useState("-");
-  const [walletNetwork, setWalletNetwork] = useState("-");
-  const [walletBalance, setWalletBalance] = useState("-");
+  const [wallet, setWalletInfo] = useState({
+    walletAddress: "-",
```

Comment:
change to undefined and display "-" if undefined

Reply:
issue fixed at #25